projects/components/src/search-box/search-box.component.ts

```diff
@@ -44,13 +49,18 @@ export class SearchBoxComponent {
   public readonly submit: EventEmitter<string> = new EventEmitter();
 
   public isFocused: boolean = false;
+  private readonly debouncedValueSubject: Subject<string> = new Subject();
+
+  public ngOnInit(): void {
+    this.debouncedValueSubject.pipe(debounceTime(this.debounceTime)).subscribe(value => this.valueChange.emit(value));
```

Comment:
what if the debounce time input changes? Also, worth unsubscribing on destroy for cleanliness if nothing else.

Reply:
Yea what was i thinking